src/content/articles/why-typescript-doesnt-follow-strict-semantic-versioning.mdx

```diff
@@ -0,0 +1,300 @@
+
+## TypeScript Does Follow (Loose) Semver
+
+Despite the aforementioned difficulties in semantic versioning TypeScript releases, the language does in fact follow a looser version of semver.
+It considers _API_ changes to be breaking:
+
+- Backwards-incompatible modifications to its Node.js APIs
+- Removals of TSConfig compiler options
+- Changes to default values of TSConfig compiler options (other than the catch-all [`strict`](https://www.typescriptlang.org/tsconfig#strict), which is explicitly noted as an exclusion)
+
+In other words, the way you might call TypeScript, such as API inputs and the general format of its outputs, are considered its stable public API.
+Nuances in how TypeScript generates its outputs, such as type checking behavior, may change in non-major releases, as with any software package.
+
+## Working with TypeScript Releases
+
+Given that TypeScript's minor releases generally change type checking behavior, updating to new TypeScript versions can sometimes be nontrivial, even practically painful.
+TypeScript's breaking changes are documented online:
+
+- Release notes are always available on the TypeScript handbook: e.g. [What's New > TypeScript 5.1](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-1.html)
+- [github.com/microsoft/TypeScript/wiki/Breaking-Changes](https://github.com/microsoft/TypeScript/wiki/Breaking-Changes): End-user changes, most notably around emitting and type checking behavior
+- [github.com/microsoft/TypeScript/wiki/API-Breaking-Changes](https://github.com/microsoft/TypeScript/wiki/API-Breaking-Changes): Changes that impact running TypeScript itself as a library or Node.js API
+
+Additionally, there are a few strategies you can take to minimize that pain.
+
+### Pinned TypeScript Versions
+
+Most software projects in the JavaScript ecosystem today use a "lockfile" such as a [`package-lock.json` (npm)](https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json), [`pnpm-lock.yaml` (pnpm)](https://pnpm.io/cli/install), and [`yarn.lock` (Yarn)](https://classic.yarnpkg.com/lang/en/docs/yarn-lock) to keep consistent versions of packages.
+A lockfile will ensure that installing dependencies in your project will always use the same versions.
+Using a lockfile is generally a good practice - especially for tools such as TypeScript that can substantially change behavior across versions.
+
+Note that when using a lockfile, you don't have to specify an exact _major.minor.patch_ version for `"typescript"` in your `package.json`.
+Your lockfile will specify an exact version of every dependency for you - generally defaulting to the latest stable version of any dependencies in your `package.json`, including `"devDependencies"`.
+
+### Intentional TypeScript Version Upgrades
+
+When you do want to update your TypeScript version, consider the following strategy:
```

Comment:
[Nitpick] This seems more like a set of steps instead of strategies.
```suggestion
When you do want to update your TypeScript version, consider the following steps:
```

Reply:
I like "strategy" :) it implies that this is a rough approach, not an explicit list of steps to take. Even if they are just steps I'd rather people see them as a strategy.